src/browser/engine/protocol.ts

```diff
@@ -1,58 +1,81 @@
 import { app, protocol } from 'electron';
-import { readFile } from 'fs';
+import { readFileSync, createReadStream } from 'fs';
+import stream from 'stream';
 import { lookup } from 'mime-types';
 import { join } from 'path';
 import { parse } from 'url';
+import ECx from './api';
 
 import { Protocol } from '../../common';
 import { protocolAsScheme } from '../../common/utils';
 import { getExtensionById } from '../chrome-extension';
 
+// tslint:disable-next-line: max-line-length
+const defaultContentSecurityPolicy = 'script-src \'self\' blob: filesystem: chrome-extension-resource:; object-src \'self\' blob: filesystem:;';
+
 protocol.registerStandardSchemes(
   [protocolAsScheme(Protocol.Extension)],
   { secure: true }
 );
 
-const protocolHandler = (
+const protocolHandler = async (
   { url }: Electron.RegisterBufferProtocolRequest,
   callback: Function
 ) => {
   const { hostname, pathname } = parse(url);
   if (!hostname || !pathname) return callback();
 
   const extension = getExtensionById(hostname);
-
   if (!extension) return callback();
 
   const { src, backgroundPage: { name, html } } = extension;
+  const headers = {};
+
+  // Hack to get extension ID
+  const splitted = src.split('/');
+  const ecxId = splitted[splitted.length - 2];
+
+  // Set Content Security Policy
+  if (ECx.isLoaded(ecxId)) {
+    const { location: { path } } = await ECx.get(ecxId);
 
+    const manifestPath = join(path, 'manifest.json');
+    const manifest = await readFileSync(manifestPath, 'utf-8');
+
+    const manifestContentSecurityPolicy = JSON.parse(manifest).content_security_policy;
+    const contentSecurityPolicy = manifestContentSecurityPolicy ? manifestContentSecurityPolicy : defaultContentSecurityPolicy;
+
+    headers['content-security-policy'] = contentSecurityPolicy;
+  }
+
+  // ?
```

Comment:
I wanted to add a comment about this use case, but couldn't figure out what it was exactly and how to describe it.

Reply:
link to https://cs.chromium.org/chromium/src/extensions/common/manifest_handlers/csp_info.cc?l=31

Reply:
Little misunderstanding about the target, but the comment is updated anyway 👍